src/path.rs

```diff
@@ -510,26 +514,31 @@ impl<'a> Iterator for Components<'a> {
         }
 
         if self.offset == 0 && path[0] == Path::SEPARATOR {
-            self.offset += 1;
+            // OK to unwrap: `offset` is 0.
+            self.offset = self.offset.checked_add(1).unwrap();
```

Comment:
if offset is 0 and we're adding a constant, why not `self.offset = 1`?

Reply:
Why not indeed! 🤦 Fixed.